open-apparel-registry/open-apparel-registry
# Add selected facility marker

## Overview

This PR changes the marker for a facility when it is selected, primarily by detecting the selected facility from the react-router details path.

One additional change here is that routes like http://localhost:6543/facilities/CN2019084R04PD8?name=shoe no longer load both the facility for the details page and the list of facilities for the search automatically on first arriving at the app from a URL entered in the browser. The details page still loads, but I updated the code to prevent the other facilities from loading, too, in order to simplify the number of cases that had to be handled for the imperative code set up around adjusting the marker styling.

Third commit is unrelated but fixes a linter complaint which I think issues from how react-router parses memoized components in the version we're using.

Connects #347

## Demo

![regular-searches](https://user-images.githubusercontent.com/4165523/54944639-5670a180-4f0a-11e9-9264-fc92ff95f73a.gif)
![visit-facility-directly](https://user-images.githubusercontent.com/4165523/54944640-57093800-4f0a-11e9-9954-5a6f9631a0c6.gif)
![multiple-faciltiies](https://user-images.githubusercontent.com/4165523/54944641-57093800-4f0a-11e9-8429-a03c75582a72.gif)

## Testing

- serve this branch then search for some facilities
- verify that selecting a facility changes its marker to selected and also unselects other facilities
- verify that you can navigate directly to a details page and see the selected facility marker
- verify that multiple facilities at the same point show the selected maker when they are selected
